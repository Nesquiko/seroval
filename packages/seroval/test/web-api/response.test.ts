import { describe, it, expect } from 'vitest';
import {
  crossSerializeAsync,
  crossSerializeStream,
  deserialize,
  fromJSON,
  serializeAsync,
  toJSONAsync,
} from '../../src';

const EXAMPLE_BODY = 'Hello World!';

describe('Response', () => {
  describe('serializeAsync', () => {
    it('supports Response', async () => {
      const example = new Response(EXAMPLE_BODY);
      const result = await serializeAsync(example);
      expect(result).toMatchSnapshot();
      const back = await deserialize<Promise<Response>>(result);
      expect(back).toBeInstanceOf(Response);
      expect(await back.text()).toBe(await example.text());
    });
  });
  describe('toJSONAsync', () => {
    it('supports Response', async () => {
      const example = new Response(EXAMPLE_BODY);
      const result = await toJSONAsync(example);
      expect(JSON.stringify(result)).toMatchSnapshot();
      const back = await fromJSON<Promise<Response>>(result);
      expect(back).toBeInstanceOf(Response);
      expect(await back.text()).toBe(await example.text());
    });
  });
  describe('crossSerializeAsync', () => {
    it('supports Blob', async () => {
      const example = new Response(EXAMPLE_BODY);
      const result = await crossSerializeAsync(example);
      expect(result).toMatchSnapshot();
    });
    describe('scoped', () => {
      it('supports Blob', async () => {
        const example = new Response(EXAMPLE_BODY);
        const result = await crossSerializeAsync(example, { scopeId: 'example' });
        expect(result).toMatchSnapshot();
      });
    });
  });

  describe('crossSerializeStream', () => {
    it('supports Response', async () => new Promise<void>((done) => {
      const example = new Response(EXAMPLE_BODY);
      crossSerializeStream(example, {
        onSerialize(data) {
          expect(data).toMatchSnapshot();
        },
        onDone() {
          done();
        },
      });
    }));
    describe('scoped', () => {
      it('supports Response', async () => new Promise<void>((done) => {
        const example = new Response(EXAMPLE_BODY);
        crossSerializeStream(example, {
          scopeId: 'example',
          onSerialize(data) {
            expect(data).toMatchSnapshot();
          },
          onDone() {
            done();
          },
        });
      }));
    });
  });
});
