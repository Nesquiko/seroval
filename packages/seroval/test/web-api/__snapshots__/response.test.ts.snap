// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Response > crossSerializeAsync > scoped > supports Blob 1`] = `"($R=>$R[0]=new Response($R[1]=new Uint8Array([72,101,108,108,111,32,87,111,114,108,100,33]).buffer,$R[2]={headers:$R[3]=new Headers({\\"content-type\\":\\"text/plain;charset=UTF-8\\"}),status:200,statusText:\\"\\"}))($R[\\"example\\"])"`;

exports[`Response > crossSerializeAsync > supports Blob 1`] = `"$R[0]=new Response($R[1]=new Uint8Array([72,101,108,108,111,32,87,111,114,108,100,33]).buffer,$R[2]={headers:$R[3]=new Headers({\\"content-type\\":\\"text/plain;charset=UTF-8\\"}),status:200,statusText:\\"\\"})"`;

exports[`Response > crossSerializeStream > scoped > supports Response 1`] = `"($R=>$R[0]=new Response($R[1]=_$.S(),$R[2]={headers:$R[3]=new Headers({\\"content-type\\":\\"text/plain;charset=UTF-8\\"}),status:200,statusText:\\"\\"}))($R[\\"example\\"])"`;

exports[`Response > crossSerializeStream > scoped > supports Response 2`] = `"($R=>_$.Se($R[1],$R[4]=new Uint8Array($R[5]=new Uint8Array([72,101,108,108,111,32,87,111,114,108,100,33]).buffer,0,12)))($R[\\"example\\"])"`;

exports[`Response > crossSerializeStream > scoped > supports Response 3`] = `"($R=>_$.Sc($R[1]))($R[\\"example\\"])"`;

exports[`Response > crossSerializeStream > supports Response 1`] = `"$R[0]=new Response($R[1]=_$.S(),$R[2]={headers:$R[3]=new Headers({\\"content-type\\":\\"text/plain;charset=UTF-8\\"}),status:200,statusText:\\"\\"})"`;

exports[`Response > crossSerializeStream > supports Response 2`] = `"_$.Se($R[1],$R[4]=new Uint8Array($R[5]=new Uint8Array([72,101,108,108,111,32,87,111,114,108,100,33]).buffer,0,12))"`;

exports[`Response > crossSerializeStream > supports Response 3`] = `"_$.Sc($R[1])"`;

exports[`Response > serializeAsync > supports Response 1`] = `"new Response(new Uint8Array([72,101,108,108,111,32,87,111,114,108,100,33]).buffer,{headers:new Headers({\\"content-type\\":\\"text/plain;charset=UTF-8\\"}),status:200,statusText:\\"\\"})"`;

exports[`Response > toCrossJSONStream > supports Response 1`] = `"{\\"t\\":36,\\"i\\":0,\\"a\\":[{\\"t\\":31,\\"i\\":1},{\\"t\\":10,\\"i\\":2,\\"p\\":{\\"k\\":[\\"headers\\",\\"status\\",\\"statusText\\"],\\"v\\":[{\\"t\\":25,\\"i\\":3,\\"e\\":{\\"k\\":[\\"content-type\\"],\\"v\\":[{\\"t\\":1,\\"s\\":\\"text/plain;charset=UTF-8\\"}],\\"s\\":1}},{\\"t\\":0,\\"s\\":200},{\\"t\\":1,\\"s\\":\\"\\"}],\\"s\\":3},\\"o\\":0}]}"`;

exports[`Response > toCrossJSONStream > supports Response 2`] = `"{\\"t\\":32,\\"i\\":1,\\"f\\":{\\"t\\":15,\\"i\\":4,\\"l\\":12,\\"c\\":\\"Uint8Array\\",\\"f\\":{\\"t\\":21,\\"i\\":5,\\"s\\":[72,101,108,108,111,32,87,111,114,108,100,33]},\\"b\\":0}}"`;

exports[`Response > toCrossJSONStream > supports Response 3`] = `"{\\"t\\":33,\\"i\\":1}"`;

exports[`Response > toJSONAsync > supports Response 1`] = `"{\\"t\\":{\\"t\\":36,\\"i\\":0,\\"a\\":[{\\"t\\":21,\\"i\\":1,\\"s\\":[72,101,108,108,111,32,87,111,114,108,100,33]},{\\"t\\":10,\\"i\\":2,\\"p\\":{\\"k\\":[\\"headers\\",\\"status\\",\\"statusText\\"],\\"v\\":[{\\"t\\":25,\\"i\\":3,\\"e\\":{\\"k\\":[\\"content-type\\"],\\"v\\":[{\\"t\\":1,\\"s\\":\\"text/plain;charset=UTF-8\\"}],\\"s\\":1}},{\\"t\\":0,\\"s\\":200},{\\"t\\":1,\\"s\\":\\"\\"}],\\"s\\":3},\\"o\\":0}]},\\"f\\":16383,\\"m\\":[]}"`;

exports[`Response > toJSONAsync > supports Response 2`] = `"{\\"t\\":36,\\"i\\":0,\\"a\\":[{\\"t\\":21,\\"i\\":1,\\"s\\":[72,101,108,108,111,32,87,111,114,108,100,33]},{\\"t\\":10,\\"i\\":2,\\"p\\":{\\"k\\":[\\"headers\\",\\"status\\",\\"statusText\\"],\\"v\\":[{\\"t\\":25,\\"i\\":3,\\"e\\":{\\"k\\":[\\"content-type\\"],\\"v\\":[{\\"t\\":1,\\"s\\":\\"text/plain;charset=UTF-8\\"}],\\"s\\":1}},{\\"t\\":0,\\"s\\":200},{\\"t\\":1,\\"s\\":\\"\\"}],\\"s\\":3},\\"o\\":0}]}"`;
